components/slices/NavigationZone.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <div class="navigation-zone">
+    <div v-for="(menuItem, index) in navigationLinks" :key="`item-${index}`">
+      <pix-link
+        v-if="!menuItem.hasOwnProperty('subNavigationLinks')"
+        :field="menuItem.link"
+        class="navigation-zone__item"
+      >
+        {{ $prismic.asText(menuItem.name) }}
+      </pix-link>
+      <div v-else>
+        <button
+          :dropdown-index="`${index}`"
+          class="dropdown-toggle navigation-zone__item links-group"
+          @click="toggleDropdown(`${index}`)"
+        >
+          {{ menuItem.name }}
+          <fa v-if="showDropdown(`${index}`)" icon="angle-up" />
+          <fa v-else icon="angle-down" />
+        </button>
+        <navigation-dropdown
+          v-if="showDropdown(`${index}`)"
+          type="button"
+          :options="menuItem.subNavigationLinks"
+          :dropdown-index="`${index}`"
+          @closeNavigationDropdown="toggleDropdown(`${index}`)"
+        >
+        </navigation-dropdown>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Vue from 'vue'
+import NavigationDropdown from '@/components/NavigationDropdown'
+
+class Navigation {
+  constructor() {
+    this.links = []
+  }
+
+  addNavigationLink(navigationLink) {
+    this.links.push(navigationLink)
+  }
+
+  addSubNavigationLink(groupName, subNavigationLink) {
+    const group = this._groupAlreadyInLinks(groupName)
+    if (!group) {
+      this._createNewGroup(groupName, [subNavigationLink])
+    } else {
+      group.subNavigationLinks.push(subNavigationLink)
+    }
+  }
+
+  _groupAlreadyInLinks(groupName) {
+    return this.links.find(
+      (link) => link.name === groupName && link.subNavigationLinks
+    )
+  }
+
+  _createNewGroup(groupName, subNavigationLinks) {
+    this.links.push({
+      name: groupName,
+      subNavigationLinks,
+    })
+  }
+}
+
+export default {
+  name: 'NavigationZone',
+  components: {
+    NavigationDropdown,
+  },
+  props: {
+    slice: {
+      type: Object,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      dropdownMap: {},
+    }
+  },
+  computed: {
+    navigationLinks() {
+      const navigation = new Navigation()
+      this.slice.items.forEach((item) => {
+        if (item.group.length > 0) {
+          const groupName = item.group[0].text
+          navigation.addSubNavigationLink(groupName, item)
+        } else {
+          navigation.addNavigationLink(item)
+        }
+      })
+      return navigation.links
```

Comment:
Pourquoi est-ce qu'on a besoin d'un objet ? Il ne contient que links d'ailleurs
```suggestion
      return links
```

Reply:
J'ai trouvé plus clair d'exporter la logique d'ajout dans le tableau dans un objet